src/main/java/org/jmt/mcmt/asmdest/ASMHookTerminator.java

```diff
@@ -136,109 +210,67 @@ public static void callTick(ServerWorld serverworld, BooleanSupplier hasTimeLeft
 			}
 			return;
 		}
-		if (mcs != server) {
+
+		if (mcServer != server) {
 			LOGGER.warn("Multiple servers?");
 			GeneralConfig.disabled = true;
 			serverworld.tick(hasTimeLeft);
 			net.minecraftforge.fml.hooks.BasicEventHooks.onPostWorldTick(serverworld);
 			return;
 		} else {
-			String taskName = null;
-			if (GeneralConfig.opsTracing) {
-				 taskName =  "WorldTick: " + serverworld.toString() + "@" + serverworld.hashCode();
-				currentTasks.add(taskName);
-			}
-			String finalTaskName = taskName;
-			p.register();
-			ex.execute(() -> {
-				try {
-					currentWorlds.incrementAndGet();
-					serverworld.tick(hasTimeLeft);
-					if (GeneralConfig.disableWorldPostTick) {
-						p.register();
-						ex.execute(() -> {
-							try {
-								//ForkJoinPool.managedBlock(
-								//		new RunnableManagedBlocker(() ->  { 
-												synchronized (net.minecraftforge.fml.hooks.BasicEventHooks.class) {
-													net.minecraftforge.fml.hooks.BasicEventHooks.onPostWorldTick(serverworld);
-												}
-								//		}));
-							//} catch (InterruptedException e) {
-							//	e.printStackTrace();
-							} finally {
-								p.arriveAndDeregister();
-							}
-						});
-					} else {
+			String taskName =  "WorldTick: " + serverworld.toString() + "@" 
+					+ serverworld.func_234923_W_().func_240901_a_().toString(); // get dimension name
```

Comment:
Is this compatible with both 1.15 and 1.16?

Reply:
Yeah, I went ahead and got rid of the reflections nonsense. Maybe another time.